src/java.desktop/share/classes/javax/swing/text/DefaultCaret.java

```diff
@@ -582,10 +582,10 @@ private void adjustFocus(boolean inWindow) {
         if ((component != null) && component.isEnabled() &&
                                    component.isRequestFocusEnabled()) {
             if (inWindow) {
-                component.requestFocusInWindow();
+                component.requestFocusInWindow(FocusEvent.Cause.MOUSE_EVENT);
```

Comment:
Looks like this function is also called only in case of mouse events from mousePressed()->adjustCaretAndFocus() and mouseClicked(). Seems fine.